kornia/constants.py

```diff
@@ -21,20 +21,41 @@ def get(cls, value: Union[str, int, T]) -> T:  # type: ignore
         raise TypeError()
 
 
-class Resample(ConstantBase, Enum):
+class EnumMetaFlags(EnumMeta):
+    def __contains__(self, other: Union[str, int, T]) -> bool:
+        if type(other) == str:
+            return any([val.name == other.upper() for val in self])
+        if type(other) == int:
+            return any([val.value == other for val in self])
+        return any([val == other for val in self])
+
+    def __repr__(self):
+        return ' | '.join(f"{self.__name__}.{val.name}" for val in self)
+
+
+class Resample(ConstantBase, Enum, metaclass=EnumMetaFlags):
     NEAREST = 0
     BILINEAR = 1
     BICUBIC = 2
 
 
-class BorderType(ConstantBase, Enum):
+class BorderType(ConstantBase, Enum, metaclass=EnumMetaFlags):
     CONSTANT = 0
     REFLECT = 1
     REPLICATE = 2
     CIRCULAR = 3
 
 
-class SamplePadding(ConstantBase, Enum):
+class SamplePadding(ConstantBase, Enum, metaclass=EnumMetaFlags):
     ZEROS = 0
     BORDER = 1
     REFLECTION = 2
+
+
+class InputType(ConstantBase, Enum, metaclass=EnumMetaFlags):
+    INPUT = 0
```

Comment:
not huge fan of this naming, a bit redundant to have `InputType` and then `INPUT` referring to images, feature tensors, etc. I would suggest something like `DataType`